src/slacko.ml

```diff
@@ -765,33 +765,22 @@ let groups_list ?exclude_archived session =
       (match d |> groups_list_obj_of_yojson with
         | Result.Ok x -> `Success x.groups
         | Result.Error x -> `ParseFailure x)
-    | #bot_error
     | #parsed_auth_error as res -> res
     | _ -> `Unknown_error
 
-exception No_matches
-exception No_unique_matches
-exception Lookup_failed
-
 (* look up the id of query from results provided by the listfn *)
 let lookupk session listfn filterfn k =
   match%lwt listfn session with
-  | `Success channels -> (match List.filter filterfn channels with
-    | [] -> Lwt.fail No_matches
-    | [x] -> Lwt.return @@ k x
-    | _ -> Lwt.fail No_unique_matches)
-  | _ -> Lwt.fail Lookup_failed
+  | #parsed_auth_error as e -> Lwt.return e
+  | `Success items -> Lwt.return @@ k @@ List.filter filterfn items
```

Comment:
That ends up looking like this:
```ocaml
let filter_map f l =
  List.fold_left
    (fun xs x -> match f x with None -> xs | Some v -> v :: xs) [] l

(* look up the id of query from results provided by the listfn *)
let lookupk session (listfn : 'a listfn) notfound filterfn =
  match%lwt listfn session with
  | #parsed_auth_error as e -> Lwt.return e
  | `Success items -> match filter_map filterfn items with
    | [id] -> Lwt.return @@ `Found id
    | _ -> Lwt.return notfound

let id_of_channel session = function
  | ChannelId id -> Lwt.return @@ `Found id
  | ChannelName name ->
    let base = String.sub name 1 @@ String.length name - 1 in
    lookupk session channels_list `Channel_not_found @@ function
    | {name = n; id = ChannelId s; _} when n = base -> Some s
    | _ -> None
```

I don't think it's any better than any of the other options, but I'm happy to go with whichever you'd prefer.